_source/_posts/2022-07-20-express-prisma-tailwind.md

```diff
@@ -0,0 +1,665 @@
+
+{% include toc.md %}
+
+### Prerequisites
+
+This tutorial uses the following technologies but doesn't require any prior experience:
+
+- An IDE for JavaScript.  I will use [Visual Studio Code](https://code.visualstudio.com/) but you can use Atom, Webstorm, or any other IDE you prefer
+- [Node.js](https://nodejs.org/en/)
+- A database, one of PostgreSQL, MySQL, SQLite, SQL Server, or MongoDB.  In this tutorial you will use SQLite
+- [Okta CLI](https://cli.okta.com)
+
+If you'd like to skip the tutorial and just check out the fully built project, you can go [view it on GitHub](https://github.com/oktadev/okta-express-prisma-tailwind-example).
+
+## Create your Okta application
+
+Create a new directory for your application.  Use the `cd` command to navigate to that folder. 
+
+{% include setup/cli.md type="web" loginRedirectUri="http://localhost:3000/authorization-code/callback" logoutRedirectUri="http://localhost:3000/" %}
+
+## Create your Express Application
+
+Next, you will use the `express-generator` application generator tool to quickly scaffold your application.  Run the following command.
+
+```console
+npx express-generator --view=pug
+```
+
+Next, you will need to install your packages.  
+
+```console
+npm i @prisma/client@3.13.0
+npm i dotenv@16.0.0
+npm i passport@0.5.2
+npm i passport-openidconnect@0.1.1
+npm i express-session@1.17
+npm i -D tailwindcss@3.0.24
+npm i -D prisma@3.13.0
+```
+
+- `@prisma/client` is used to access the database from your server code
+- `dotenv` reads configuration settings from `.env` files like the one produced by the Okta CLI
+- `passport` is a middleware for Node.js that is flexible enough to handle most authentication scenarios, including Okta.  `passport-openidconnect` is a module for passport that lets you authenticate with OpenID Connect
+- `express-session` is required for passport.  Express apps use this package for session support.  You application must intiialize session support to use passport.  
+- `Tailwind CSS` is the CSS framework you will use.  If you've never used Tailwind before, you may wonder why it's a development dependency.  This is because Tailwind will dynamically build your CSS files from your views and configuration.  More on that later
+- `prisma` is the ORM you are using.  This is a dev dependency because the Prisma library handles the migrations, seed data, etc., while the `@prisma/client` package is used in your application at runtime.
+
+At this point, you will initialize Tailwind CSS and Prisma. You will configure them for usage later on.
+
+Start with Prisma and run the following command.
+
+```console
+npx prisma init
```

Comment:
Should there be a version specified on this npx command too? (or verify that it will use the version specified in package.json automatically)